src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -85,6 +88,10 @@ public class ProgrammingExerciseExportService {
 
     public static final String EXPORTED_EXERCISE_PROBLEM_STATEMENT_FILE_PREFIX = "Problem-Statement";
 
+    private static final String EMBEDDED_FILE_REGEX = "!\\[.*\\.(jpg|jpeg|png|svg|gif|pdf)] *\\(.*\\)";
```

Comment:
I don’t think this matches images properly:

- The part in `[]` is the `alt="…"` attribute of the `<img …>` HTML tag. That can by an arbitrary image description written by the instructor.
- The `(…)` is the actual image link. That can even be a full http-URL for externally hosted images. Those cannot be copied. I think we probably only care about anything that matches `/api/files/markdown/.*` here regardless of the file extension. We can probably safely assume that only allowed files could have been uploaded in the first place.
   - That same assumption then also can be made for the import.
   - That shouldn’t be a security risk, as the export/import/edit of exercises is only done by editors/instructors/admins which have to be trusted anyway to provide safe Git repository template files etc.

Reply:
Ok, so if I understand this correctly, we only care about the content in the rounded brackets.
For the export we can safely assume this because all other files are blocked anyway when someone tries to upload them. So the export doesn't contain any invalid files. Hence, the import would only contain invalid, potentially malicious files if someone actively modifies the export which is possible for sure but the same applies to the other files you mentioned.